crates/parser/src/format/mod.rs

```diff
@@ -63,7 +63,15 @@ pub fn resolve_config(
         tracing::debug!("using user-provided format: {:?}", f);
         f.clone()
     } else {
-        let inferred = determine_format(&config).ok_or_else(|| ParseError::CannotInferFormat)?;
+        let inferred = determine_format(&config).ok_or_else(|| {
+            ParseError::CannotInferFormat(
+                config
+                    .filename
+                    .as_deref()
+                    .expect("file name must be known")
```

Comment:
As originally written, the `filename` was intentionally not a required field in the config. It _might_ be the case that filename is always provided in practice, but I'm not sure. I think if we do want to require that `filename` is always defined, then we should do so by making it non-optional in the config so we can fail fast if it's not provided.

Otherwise, I'd suggest defaulting the name to `stdin` or something instead of panicking.